File: 338_counting_bits/counting_bits.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* countBits(int num, int* returnSize){
    *returnSize = num + 1;
    int *result = malloc(sizeof(int) * (num + 1));
    result[0] = 0;
    for (int i = 1; i <= num; i++) {
        result[i] = result[i >> 1] + i % 2;
    }
    return result;
}

File: 338_counting_bits/counting_bits.py
from typing import List


class Solution:
    def countBits(self, num: int) -> List[int]:
        # 方法一
        result = []
        for i in range(num + 1):
            x, count = i, 0
            while x > 0:
                x &= (x - 1)
                count += 1
            result.append(count)
        return result
